server/applicationPattern/applicationPattern/onfModel/services/ForwardingConstructAutomationServices.test.js

```diff
@@ -0,0 +1,412 @@
+  {
+    forwardingName: 'OperationUpdateBroadcast',
+    attributeList: {
+      'application-name': 'RegistryOffice',
+      'release-number': '2.0.1',
+      'old-operation-name': '/v1/register-application',
+      'new-operation-name': '/v2/register-application'
+    },
+    context: undefined
+  }
+];
+
+const ltpsForSubscription = [
+  {
+    "uuid": "ro-2-0-1-op-s-is-011",
+    "ltp-direction": "core-model-1-4:TERMINATION_DIRECTION_SOURCE",
+    "client-ltp": [
+    ],
+    "server-ltp": [
+      "ro-2-0-1-http-s-000"
+    ],
+    "layer-protocol": [
+      {
+        "local-id": "0",
+        "layer-protocol-name": "operation-server-interface-1-0:LAYER_PROTOCOL_NAME_TYPE_OPERATION_LAYER",
+        "operation-server-interface-1-0:operation-server-interface-pac": {
+          "operation-server-interface-capability": {
+            "operation-name": "/v1/relay-operation-update"
+          },
+          "operation-server-interface-configuration": {
+            "life-cycle-state": "operation-server-interface-1-0:LIFE_CYCLE_STATE_TYPE_EXPERIMENTAL",
+            "operation-key": "Operation key not yet provided."
+          }
+        }
+      }
+    ]
+  }
+]
+
+const fcForSubscription = {
+  "uuid": "ro-2-0-1-op-fc-is-010",
+  "name": [
+    {
+      "value-name": "ForwardingKind",
+      "value": "core-model-1-4:FORWARDING_KIND_TYPE_SUBSCRIPTION"
+    },
+    {
+      "value-name": "ForwardingName",
+      "value": "OperationUpdateBroadcast"
+    }
+  ],
+  "fc-port": [
+    {
+      "local-id": "000",
+      "port-direction": "core-model-1-4:PORT_DIRECTION_TYPE_MANAGEMENT",
+      "logical-termination-point": "ro-2-0-1-op-s-is-003"
+    },
+    {
+      "local-id": "001",
+      "port-direction": "core-model-1-4:PORT_DIRECTION_TYPE_MANAGEMENT",
+      "logical-termination-point": "ro-2-0-1-op-s-is-002"
+    },
+    {
+      "local-id": "100",
+      "port-direction": "core-model-1-4:PORT_DIRECTION_TYPE_INPUT",
+      "logical-termination-point": "ro-2-0-1-op-s-is-011"
+    }
+  ]
+};
+
+beforeEach(() => {
+  jest.spyOn(fileOperation, 'writeToDatabaseAsync').mockImplementation(() => true);
+  jest.spyOn(fileOperation, 'deletefromDatabaseAsync').mockImplementation(() => true);
+  ltps[0] = ltp;
+})
+
+describe("configureForwardingConstructAsync", () => {
+  test("automateForwardingConstructAsync - successful", async () => {
+    const operationServerName = '/v1/embed-yourself';
+    jest.spyOn(ControlConstruct, 'getLogicalTerminationPointListAsync').mockImplementation(() => ltps);
+    jest.spyOn(ForwardingDomain, 'getForwardingConstructForTheForwardingNameAsync').mockImplementation(() => fc);
+    jest.spyOn(event, 'dispatchEvent').mockImplementation(() => true);
+    
+    const res =  await ForwardingConstructAutomationServices.automateForwardingConstructAsync(
+      operationServerName, forwardingAutomationInputList, 'User Name', 
+      '550e8400-e29b-11d4-a716-446655440000', '1.3.1', 'Unknown value');
+    expect(res).toBe(undefined);
```

Comment:
This will never fail, since the method does not return anything, so again, nothing is tested here. What should be tested is with which values the eventDispatcher was called.